api/resources/views.py

```diff
@@ -77,3 +77,6 @@ def get_object(self):
 
     def get_permissions_proxy(self):
         return Guid.load(self.get_object().primary_resource_guid).referent
+
+    def perform_destroy(self, instance):
+        instance.delete(api_request=self.request)
```

Comment:
Should this be `instance.deleted = timezone.now()`?

Reply:
See the custom delete logic from https://github.com/CenterForOpenScience/osf.io/pull/9989 (and the tests)